src/commonMain/kotlin/app/revanced/library/installation/installer/Constants.kt

```
@@ -19,14 +19,14 @@ internal object Constants {
     const val CREATE_INSTALLATION_PATH = "$CREATE_DIR $MOUNT_PATH"
 
     const val MOUNT_APK =
-        "base_path=\"$MOUNTED_APK_PATH\" && " +
+        "base_path='$MOUNTED_APK_PATH' && " +
             "mv $TMP_FILE_PATH \$base_path && " +
             "chmod 644 \$base_path && " +
             "chown system:system \$base_path && " +
             "chcon u:object_r:apk_data_file:s0  \$base_path"
```

Could use multiline string here and in other constants so that && can be removed, the line length reduced and readability improved. Also escaping " wont be needed anymore. .trimIndent() should be used then.

---

Already did that change and then reverted it. Think current code is readable enough and `&&`-chaining has the benefit of failing fast and returning the failure code.

Single quotes for `base_path` var are ok here, as everything is expanded on java side, no need for shell-side expansion anyway.

---

> and &&-chaining has the benefit of failing fast and returning the failure code.

What do you mean? Wouldn't running the commands in a new line also return the failure code

Short multiple lines rather than a single long one is more readable, ideally this is applied here.

---

> Wouldn't running the commands in a new line also return the failure code

By default, without the `set -e` (which as discussed somewhere else I don't know whether shell on android supports), no. E.g.

```sh
command1
command2
command3
```

would return 0 if only `command2` failed.

---

Why not if command 1 or 3 failed?

---

If _only_ comman21 and/or 2 failed, then the script would succeed as its exit code would be that of command3's - meaning successful.

---

Doesn't command1 failing exit the script? From what I know if a line fails the lines below arent run.

---

Not familiar with android, but in dash or bash, with default settings - no. That's what the `set -e` is for.